lib/cookie.c

```diff
@@ -239,6 +240,49 @@ static bool pathmatch(const char *cookie_path, const char *request_uri)
   return ret;
 }
 
+/*
+ * Return the top-level domain, for optimal hashing.
+ */
+static const char *get_top_domain(const char * const in, unsigned *outlen)
+{
+  unsigned len;
+  const char *first, *last;
+
+  if(!in)
+    return NULL;
+
+  len = strlen(in);
+  first = memchr(in, '.', len);
+  last = memrchr(in, '.', len);
+
+  if(outlen)
+    *outlen = len;
+
+  if(first == last)
+    return in;
+
+  first = memrchr(in, '.', last - in - 1);
+  if(outlen)
+    *outlen = len - (first - in) - 1;
+
+  return first + 1;
+}
+
+/*
+ * Hash this domain.
+ */
+static size_t cookiehash(const char * const in)
```

Comment:
... this also uses 'in' without much explanation. Use a better name or add a comment saying what it is?